projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -82,15 +116,41 @@ export class CartesianLegend {
     const legendEntry = select<EnterElement, Series<{}>>(element).append('div').classed('legend-entry', true);
 
     this.appendLegendSymbol(legendEntry);
-
     legendEntry
       .append('span')
       .classed('legend-text', true)
-      .text(series => series.name);
+      .text(series => series.name)
+      .on('click', series => this.updateActiveSeries(series));
```

Comment:
If there's only one series, can we avoid adding this? We don't need selection mode for a single series graph (which is many of them).